data_steward/cdr_cleaner/cleaning_rules/identifying_field_suppression.py

```diff
@@ -102,24 +102,25 @@ def get_query_specs(self, *args, **keyword_args) -> query_spec_list:
             are optional but the query is required.
         """
         queries = []
+        schemas = {}
+        for table in self.affected_tables:
+            schemas[table] = fields_for(table)
```

Comment:
One last suggestion, instead of looping through the list of affected tables twice, you could just do one loop.

I think you've got :
```python
        for table in self.affected_tables:
            schemas[table] = fields_for(table)
        for table in self.affected_tables:
            statements = []
            for item in schemas[table]:
```
And I think it could be simplified to:
```python
        for table in self.affected_tables:
            statements = []
            schema = fields_for(table)
            for item in schema:
```
Or is there a design decision that I am not understanding for looping through this list twice?

Reply:
Thanks for catching this, it is an artifact of moving it from the setup function.